dev-tools/packaging/packages.yml

```diff
@@ -62,19 +62,19 @@ shared:
       /etc/init.d/{{.BeatServiceName}}:
         template: '{{ elastic_beats_dir }}/dev-tools/packaging/templates/{{.PackageType}}/elastic-agent.init.sh.tmpl'
         mode: 0755
-      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{ commit_short }}/{{.BeatName}}{{.BinaryExt}}:
+      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{agent_package_version}}{{snapshot_suffix}}-{{ commit_short }}/{{.BeatName}}{{.BinaryExt}}:
         source: build/golang-crossbuild/{{.BeatName}}-{{.GOOS}}-{{.Platform.Arch}}{{.BinaryExt}}
         mode: 0755
-      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{ commit_short }}/package.version:
+      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{agent_package_version}}{{snapshot_suffix}}-{{ commit_short }}/package.version:
         content: >
           {{ agent_package_version }}
         mode: 0644
-      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{ commit_short }}/components:
+      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{agent_package_version}}{{snapshot_suffix}}-{{ commit_short }}/components:
         source: '{{.AgentDropPath}}/{{.GOOS}}-{{.AgentArchName}}.tar.gz/'
         mode: 0755
         config_mode: 0644
         skip_on_missing: true
-      /var/lib/{{.BeatName}}/data/{{.BeatName}}-{{ commit_short }}/manifest.yaml:
```

Comment:
Why do we make these changes for the deb/rpm spec but not others, e.g. MacOS? Or, to put it differently, if we made the corresponding changes in other specs would we still need the install code changes in this PR?

I'm sure there's a valid explanation for this difference; I'm just not understanding it yet.

Reply:
Never mind, ignore this question.  It looks like the MacOS section is not even being used in packaging.  I tried deleting that section, repackaged Agent, installed it, and it worked fine.  I'll make a separate PR to just remove that section so I don't get confused by it again in the future.

Reply:
https://github.com/elastic/elastic-agent/pull/4182

Reply:
Didn't check yet if we are actually using that MacOS package specification, will do that later.
As for why we are only changing this on .deb and .rpm: we install those files without calling the elastic-agent install command but we rely on CLI utilities to so that.

We want to have the version in the installation path when installing the agent so we need to change the directory structure at packaging time.

We use tar.gz and .zip packages for both install and upgrade: as discussed in a recent RFC we want to maintain the package directory structure with `elastic-agent-<hash>` for backward compatibility with previous versions of agent (hence this change is not applied to those package types) and we handle the remapping to `elastic-agent-<version>-<hash>` within the `install` and `upgrade` commands